### PR TITLE
Update governance ledger model

### PR DESCRIPTION
**Pull request by andrew-lee-work**
Update the governance ledger model to the Rust implementation as of Feb 2024.

* Deviations from the implementation are noted in a comment.
* Some bug fixes are included.

### CODE REVIEW

File: tla/governance-ledger/Governance.tla [hunk 1]
```diff
@@ -649,13 +621,42 @@ process ( Merge_Neurons \in Merge_Neurons_Process_Ids )
...
+                update_fees(source_neuron_id, fees_amount);
+                send_request(self, OP_TRANSFER,
+                        transfer(neuron[source_neuron_id].account,
+                            neuron[target_neuron_id].account,
+                            (neuron[source_neuron_id].cached_stake - neuron[source_neuron_id].fees) - TRANSACTION_FEE,
+                            TRANSACTION_FEE));
+            };
+        };
+
+    MergeNeurons3:
```
Comment (oggy-dfin) [inline tla/governance-ledger/Governance.tla:660]: The `goto MergeNeuron5` below should be `goto MergeNeuron6` now I believe?

File: tla/governance-ledger/Governance.tla [hunk 2]
```diff
@@ -664,13 +665,13 @@ process ( Merge_Neurons \in Merge_Neurons_Process_Ids )
                 send_request(self, OP_QUERY_BALANCE, balance_query(neuron[target_neuron_id].account));
             };
         };
-    MergeNeurons3:
+    MergeNeurons4:
```
Comment (oggy-dfin) [inline tla/governance-ledger/Governance.tla:669]: I can't find the balance query stuff in the code any more, but I find that the code has also unfortunately become a lot more difficult to understand; could you point me to where this query happens?

Comment (andrew-lee-work) [reply tla/governance-ledger/Governance.tla:669]: The implementation does not query ledger, and IIRC the implementation did not previously do that either, especially after already doing the transfer. I think this is an intentional divergence from the implementation in the TLA model. The additional query seems to be added to catch a violation of an invariant.

File: tla/governance-ledger/Governance.tla [hunk 3]
```diff
@@ -95,12 +95,29 @@ Apalache caveats to keep in mind:
...
+\* TODO: this documentation should be updated to not refer to Merge_Maturity_Of_Neuron
 The model deviates from the Rust implementation by (among other things) employing some optimizations.
 In particular, validity checks (e.g., in Merge_Maturity_Of_Neuron) over method parameters are often 
 moved to the start of the action, such that the choice of the parameters is restricted immediately, 
 rather adding an "await" at the same point as where the implementation performs its checks.
 This shortens the time needed by TLC to figure out that an action is disabled in a particular state.
 
+
+***********
+Deviations
```
Comment (oggy-dfin) [inline tla/governance-ledger/Governance.tla:107]: Could you add the merge_neuron differences here?

**Review by oggy-dfin — APPROVED** [review]
Pre-approving, please add a note about merge neuron to the deviations

File: tla/governance-ledger/Governance.tla (hunk 3 above)
Comment (andrew-lee-work) [reply tla/governance-ledger/Governance.tla:107]: done

**Pull request merged**